ui/app/src/components/PublishDialog/PublishItemsView.tsx

```diff
@@ -0,0 +1,266 @@
+  dependencyTypeMap?: DependencyMap;
+  onCheckboxChange?: (e: React.ChangeEvent<HTMLInputElement>, checked: boolean, path: string) => void;
+  selectedDependencies?: string[];
+}) {
+  const { itemMap, node, onMenuClick, dependencyTypeMap, onCheckboxChange, selectedDependencies } = props;
+  const isItem = Boolean(itemMap[node.path]);
+  const isDependency = Boolean(dependencyTypeMap?.[node.path]);
+  const isSoft = dependencyTypeMap?.[node.path] === 'soft';
+  return (
+    <TreeItem
+      key={node.path}
+      itemId={node.path}
+      data-is-item={isItem}
+      label={
+        isItem ? (
+          <Box display="flex" justifyContent="space-between" alignItems="center">
+            <div>
+              <Box display="flex">
+                <ItemDisplay
+                  item={itemMap[node.path]}
+                  showNavigableAsLinks={false}
+                  showWorkflowState={false}
+                  sx={{ mr: 1 }}
+                />
+                {isDependency && <DependencyChip type={dependencyTypeMap[node.path]} />}
+              </Box>
+              <Typography
+                component="div"
+                variant="body2"
+                color="text.secondary"
+                children={node.path}
+                title={node.path}
+                noWrap
+              />
+            </div>
+            <Box display="flex">
+              <IconButton
+                className="tree-item-more-section"
+                onClick={(e) => {
+                  e.stopPropagation();
+                  onMenuClick?.(e, node.path);
+                }}
+              >
+                <MoreVertRounded />
+              </IconButton>
+              {isSoft && (
+                <Checkbox
+                  size="small"
+                  checked={selectedDependencies?.includes(node.path)}
+                  onClick={(e) => e.stopPropagation()}
+                  onChange={(e, checked) => {
+                    onCheckboxChange?.(e, checked, node.path);
+                  }}
+                />
+              )}
+            </Box>
+          </Box>
+        ) : (
+          // TODO: Add folder icon
+          <span title={node.path}>{node.label}</span>
+        )
+      }
+      children={
+        node.children?.length === 0
+          ? undefined
+          : node.children.map((child) =>
+              renderTreeNode({
+                itemMap,
+                node: child,
+                dependencyTypeMap,
+                onMenuClick,
+                onCheckboxChange,
+                selectedDependencies
+              })
+            )
+      }
+    />
+  );
+}
+
+export function PublishItemsView(props: PublishItemsProps) {
+  const {
+    itemMap,
+    isTreeView,
+    setIsTreeView,
+    expandedPaths,
+    itemsAndDependenciesPaths,
+    dependencyTypeMap = {},
+    selectedDependenciesPaths = [],
+    selectedDependenciesMap = {},
+    trees,
+    setExpandedPaths,
+    onMenuClick,
+    onCheckboxChange
```

Comment:
Assess if there are props not required on the consumer component that can be moved to local state (e.g. isTreeView? expandedPaths?)